<launch>
  <!--include file="$(find trim_apriltag)/launch/src/cam_c270_1.launch" /-->
  <node name="play" pkg="rosbag" type="play" args="$(find trim_apriltag)/rosbag/2tag.bag --loop"/>

  <arg name="launch_prefix" default="" /> <!-- set to value="gdbserver localhost:10000" for remote debugging -->
  <arg name="node_namespace" default="apriltag_ros_continuous_node" />
  <arg name="camera_name" default="/usb_cam" />
  <arg name="camera_frame" default="usb_cam" />
  <arg name="image_topic" default="image_raw" />



  <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg node_namespace)" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)" >
    <remap from="image_rect" to="/masking_image" />
    <remap from="camera_info" to="/masking_info" />

    <param name="camera_frame" type="str" value="$(arg camera_frame)" />
    <param name="publish_tag_detections_image" type="bool" value="true" />
  </node>


  <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="/tag_detections_image"/>
    <param name="autosize" value="true" />
  </node>


  <node name="masking_image" pkg="trim_apriltag" type="2tag_trim_include.py" output="screen"/>


  <!-- Set parameters -->
  <rosparam command="load" file="$(find trim_apriltag)/config/settings.yaml" ns="$(arg node_namespace)" />
  <rosparam command="load" file="$(find trim_apriltag)/config/tags.yaml" ns="$(arg node_namespace)" />

</launch>
